puppet-checks/src/main/java/com/iadams/sonarqube/puppet/checks/FileModeCheck.java

```diff
@@ -60,22 +63,36 @@ public void init() {
 
   @Override
   public void visitNode(AstNode node) {
-    if(node.getTokenValue().equals("file")) {
+    if ("file".equals(node.getTokenValue())) {
       for (AstNode body : node.getDescendants(PuppetGrammar.RESOURCE_INST)) {
```

Comment:
@racodond Yes, i have modelled the grammar almost verbatim after the ruby implementation, i'm happy to add skips where they add value (I wasn't sure sure if there are any best practices around maintaining the AST vs easier checks etc).